### PR TITLE
OPENEUROPA-2044: Integrate global banner from webtools.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

A Europa-wide banner certifying the website is officially owner by the EU. It comes with different styles and configuration options.

Integrate webtools widget.
Add configuration form.
Tests.
### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_webtools_globan/oe_webtools_globan.module [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+use Drupal\Core\Asset\AttachedAssetsInterface;
+use Drupal\Core\Cache\CacheableMetadata;
+
+/**
+ * Implements hook_page_attachments().
+ *
+ * Attach smart loader.
+ */
+function oe_webtools_globan_page_attachments(array &$attachments): void {
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.module]: Please check that we do actually need this cache meta

File: modules/oe_webtools_globan/oe_webtools_globan.module [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  if (\Drupal::service('router.admin_context')->isAdminRoute()) {
+    return;
+  }
+
+  /** @var \Drupal\Core\Config\ConfigFactoryInterface $config */
+  $config = \Drupal::service('config.factory')->getEditable('oe_webtools_globan.settings');
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.module]: Dont use getEditable

File: modules/oe_webtools_globan/oe_webtools_globan.module [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    $globan_param = $config->get('display_eu_flag') ?? '1';
+    $globan_param .= $config->get('background_theme') ?? '1';
+    $globan_param .= $config->get('eu_institutions_links') ?? '1';
+
+    $lang = '';
+    if ($config->get('override_page_lang')) {
+      $lang = '&lang=' . $config->get('override_page_lang');
+    }
+
+    $javascript['//europa.eu/webtools/load.js']['data'] .= '?globan=' . $globan_param . $lang;
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.module]: Fix the URL building to use the API.

File: modules/oe_webtools_globan/js/globan-livepreview.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+(function ($, Drupal, window) {
```
Comment (upchuk) [inline modules/oe_webtools_globan/js/globan-livepreview.js]: Remove this and the preview.

File: modules/oe_webtools_globan/src/Form/WebtoolsGlobanSettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $config = $this->config('oe_webtools_globan.settings');
+
+    $form['globan_settings'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Global banner settings'),
+    ];
+    $form['globan_settings']['livepreview'] = [
```
Comment (upchuk) [inline modules/oe_webtools_globan/src/Form/WebtoolsGlobanSettingsForm.php]: Remove preview

File: modules/oe_webtools_globan/src/Form/WebtoolsGlobanSettingsForm.php [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      '#type' => 'fieldset',
+      '#title' => $this->t('Check appearance'),
+      '#markup' => '<div id="globan-preview"></div>',
+      '#description' => $this->t('Link for testing appearance of global banner.'),
+    ];
+    $form['globan_settings']['display_eu_flag'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Display EU flag'),
+      '#options' => [
+        '0' => $this->t('No - hide flag'),
```
Comment (upchuk) [inline modules/oe_webtools_globan/src/Form/WebtoolsGlobanSettingsForm.php]: This should be int and in the .schema it should be a boolean not string. It's a flag. Same for all the other flags.

File: tests/features/global-banner.feature [hunk 7]
```diff
@@ -0,0 +1,30 @@
+@api
```
Comment (upchuk) [inline tests/features/global-banner.feature]: Please test also the caching (log in and out)

File: modules/oe_webtools_globan/oe_webtools_globan.module (hunk 1 above)
Comment (sergepavle) [reply modules/oe_webtools_globan/oe_webtools_globan.module]: Confirmed. Without this cache meta, cache invalidation doesn't work correctly.

File: modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+oe_webtools_globan.settings:
+  type: config_object
+  label: 'Webtools Globan settings'
+  mapping:
+    display_eu_flag:
+      type: boolean
+      label: 'Display EU flag'
+      description: 'Enable or disable the EU flag icon in the Global Banner.'
+    background_theme:
```
Comment (upchuk) [inline modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml]: Can we change this to something that makes more sense? Like to store a string `light` or `dark`? It's not really a boolean if we are choosing between two values that are not really on/off, yes/no, etc. Especially if the config name is simply called "background_theme"

File: modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+      type: boolean
+      label: 'Background theme'
+      description: 'Select whether the Global Banner theme is light or dark.'
+    eu_institutions_links:
```
Comment (upchuk) [inline modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml]: `display_eu_institutions_links`

File: modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+      type: boolean
+      label: 'See all EU Institutions and bodies'
+      description: 'Show or hide a link to all EU institutions and Bodies.'
+    override_page_lang:
+      type: string
+      label: 'Override page language'
+      description: 'Use this option if you want to display a language, different from current page.'
```
Comment (upchuk) [inline modules/oe_webtools_globan/config/schema/oe_webtools_globan.schema.yml]: `Use this option if you want to display in a language that is different from current page language.`

File: modules/oe_webtools_globan/oe_webtools_globan.info.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+name: OpenEuropa Webtools Globan
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.info.yml]: OpenEuropa Webtools Global Banner

File: modules/oe_webtools_globan/oe_webtools_globan.links.menu.yml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+oe_webtools_globan.settings:
+  title: Webtools Globan
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.links.menu.yml]: `Webtools Global Banner`

File: modules/oe_webtools_globan/oe_webtools_globan.module [hunk 13]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal hooks.
```
Comment (upchuk) [inline modules/oe_webtools_globan/oe_webtools_globan.module]: `The Webtools Global Banner module file.`

**Pull request merged**